substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -1168,7 +1168,7 @@ pub mod pallet {
             let node_serial_number = Self::get_serial_number(serial_number)?;
 
             // If the resources on a certified node changed, reset the certification level to DIY
-            if stored_node.resources != node_resources
+            if !node_resources.has_changed(stored_node.resources, 1).is_ok()
```

Comment:
The function is a bit confusing indeed. Can we mabye make it static? So that you can do something like this instead:
```
if Resources::has_changed(resources_before, resources_after, tolerance)
```
And why not make it return a boolean?